docs/architecture-decisions/2023-06-06_frontend-templates.md

```diff
@@ -0,0 +1,133 @@
+
+## Requirements
+
+- **Modularity**: The dApp templating feature should follow guiding AlgoKit principles and expand on approaches already utilized in existing smart contract templates feature. This implies that giving developers flexibility to mix and match different `smart contract` templates with `frontend` templates should serve as key consideration.
+- **Maintainability**: The dApp templating feature should be easy to maintain and extend. This implies that the feature should be implemented in a way that allows for easy addition of new templates and/or modification of existing ones as the complexity and variety of templates scale.
+- **Seamless onramp**: The dApp templating feature should provide a seamless onramp for developers to get started with dApp development. This implies that the feature should provide a simple and intuitive way to get started with dApp development and deployment. Providing developers a choice on whether they want more flexibility or rely on default recommended practices.
+
+## Principles
+
+- [AlgoKit Guiding Principles](../../docs/algokit.md#Guiding-Principles) - all of the aforementioned requirements should be met in a way that is consistent with the guiding principles of AlgoKit or attempt to find a balanced trade of between the principles that satisfies the requirements.
+
+## Options
+
+Expanding AlgoKit templating capabilities is a non-trivial task that requires careful consideration of the trade-offs between the different approaches. The following section outlines the different options that were considered and the trade-offs associated with each of them.
+
+> Refer to git history to see old revisions of the document. After further discussions with the team we are now aligned on the separate repositories approach. Hence the sections below are updated to reflect the new approach.
+
+## The proposal
+
+The proposal consists of 2 main parts. On a high level, the ideas revolve around giving developers a choice on how they want to couple AlgoKit backend and frontend templates if they like to use custom templates. The other is focused on providing an official full-stack template that bundles both backend and frontend templates together, while also serving as an example for template builders on how to efficiently couple their backend and frontend templates.
```

Comment:
Minor "couple their backend and frontend templates." - not just frontend and backend, any combination of multiple templates they may want.

Reply:
Should be addressed now. Feel free to un-resolve if you think there are still items to modify.